insecure/attacknetwork/attackNetwork.go

```diff
@@ -0,0 +1,244 @@
+			return nil
+		default:
+			msg, err := stream.Recv()
+			if err == io.EOF {
+				a.logger.Info().Msg("attack network closed processing stream")
+				return stream.SendAndClose(&empty.Empty{})
+			}
+			if err != nil {
+				return fmt.Errorf("could not read corrupted node's stream: %w", err)
+			}
+
+			if err = a.processMessageFromCorruptedNode(msg); err != nil {
+				a.logger.Fatal().Err(err).Msg("could not process message of corrupted node")
+				return stream.SendAndClose(&empty.Empty{})
+			}
+		}
+	}
+}
+
+// processMessageFromCorruptedNode processes incoming messages arrived from corruptible conduits by passing them
+// to the orchestrator.
+func (a *AttackNetwork) processMessageFromCorruptedNode(message *insecure.Message) error {
+	event, err := a.codec.Decode(message.Payload)
+	if err != nil {
+		return fmt.Errorf("could not decode observed payload: %w", err)
+	}
+
+	sender, err := flow.ByteSliceToId(message.OriginID)
+	if err != nil {
+		return fmt.Errorf("could not convert origin id to flow identifier: %w", err)
+	}
+
+	targetIds, err := flow.ByteSlicesToIds(message.TargetIDs)
+	if err != nil {
+		return fmt.Errorf("could not convert target ids to flow identifiers: %w", err)
+	}
+
+	err = a.orchestrator.HandleEventFromCorruptedNode(&insecure.Event{
+		CorruptedId:       sender,
+		Channel:           network.Channel(message.ChannelID),
+		FlowProtocolEvent: event,
+		Protocol:          message.Protocol,
+		TargetNum:         message.Targets,
+		TargetIds:         targetIds,
+	})
+	if err != nil {
+		return fmt.Errorf("could not handle event by orchestrator: %w", err)
+	}
+
+	return nil
+}
+
+// Send enforces dissemination of given event via its encapsulated corrupted node networking layer through the Flow network
+func (a *AttackNetwork) Send(event *insecure.Event) error {
+
+	connection, ok := a.corruptedConnections[event.CorruptedId]
+	if !ok {
+		return fmt.Errorf("no connection available for corrupted conduit factory to node %x: ", event.CorruptedId)
+	}
+
+	msg, err := a.eventToMessage(event.CorruptedId, event.FlowProtocolEvent, event.Channel, event.Protocol, event.TargetNum, event.TargetIds...)
+	if err != nil {
+		return fmt.Errorf("could not convert event to message: %w", err)
+	}
+
+	err = connection.SendMessage(msg)
+	if err != nil {
+		return fmt.Errorf("could not sent event to corrupted node: %w", err)
+	}
+
+	return nil
+}
+
+// eventToMessage converts the given application layer event to a protobuf message that is meant to be sent to the corrupted node.
+func (a *AttackNetwork) eventToMessage(corruptedId flow.Identifier,
+	event interface{},
+	channel network.Channel,
+	protocol insecure.Protocol,
+	num uint32,
+	targetIds ...flow.Identifier) (*insecure.Message, error) {
+
+	payload, err := a.codec.Encode(event)
+	if err != nil {
+		return nil, fmt.Errorf("could not encode event: %w", err)
+	}
+
+	return &insecure.Message{
+		ChannelID: channel.String(),
+		OriginID:  corruptedId[:],
+		Targets:   num,
```

Comment:
nit: should this be `TargetCount`? `Targets` seems to imply it's a list of targets

Reply:
fixed in https://github.com/onflow/flow-go/pull/2144/commits/9049d4ffec6c0e6a9f033366587586d029fec535